src/main/java/com/zerobase/mentalgrowhdiary/service/CounselorService.java

```diff
@@ -67,4 +71,24 @@ public List<CounselorResponseDto> searchCounselors(String name, String feature,
         return counselors.stream().map(
                 counselor -> CounselorResponseDto.fromEntity(counselor,objectMapper)).toList();
     }
+
+    @Transactional(readOnly = true)
+    public List<ClientDiaryResponseDto> getClientDiaries(String counselorName) {
+
+
+        List<Diary> diaries = userCounselorRepository.findClientDiaryByConditions(counselorName);
+
+        return diaries.stream()
+                .map(diary -> {
+                    return new ClientDiaryResponseDto(
+                            diary.getUser().getUserId(),
+                            diary.getUser().getUsername(),
+                            diary.getDiaryId(),
+                            diary.getDiaryContent(),
+                            diary.getDiaryDate()
+                    );
+                })
+                .toList();
+    }
```

Comment:
불필요한 선언 없이 다음과 같이 해도 좋을 것 같네요
```suggestion
        return userCounselorRepository.findClientDiaryByConditions(counselorName).stream()
                .map(diary -> {
                    return new ClientDiaryResponseDto(
                            diary.getUser().getUserId(),
                            diary.getUser().getUsername(),
                            diary.getDiaryId(),
                            diary.getDiaryContent(),
                            diary.getDiaryDate()
                    );
                })
                .toList();
    }
```